src/tools/rbd/action/Snap.cc

```diff
@@ -169,7 +169,10 @@ int do_unprotect_snap(librbd::Image& image, const char *snapname)
 
 int do_set_limit(librbd::Image& image, uint64_t limit)
 {
-  return image.snap_set_limit(limit);
+  int r = image.snap_set_limit(limit);
+  if (r == -EINVAL)
+      std::cerr << "rbd: the image has more snaps than the new limit set " << std::endl;
```

Comment:
No, EINVAL is a good choice. but the error message should not printed here. move it into snap_set_limit.